test/test_ffs.cpp

```diff
@@ -74,31 +75,41 @@ TORRENT_TEST(clz)
 		std::fprintf(stderr, "%s\n", t.first);
 		std::uint32_t buf[5];
 		to_binary(t.first, buf);
-		TEST_EQUAL(aux::clz_sw({buf, 5}), t.second);
-		TEST_EQUAL(aux::clz_hw({buf, 5}), t.second);
-		TEST_EQUAL(aux::clz({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros_sw({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros_hw({buf, 5}), t.second);
+		TEST_EQUAL(aux::count_leading_zeros({buf, 5}), t.second);
 	}
 }
 
-TORRENT_TEST(flz_u32)
+TORRENT_TEST(find_last_zero_u32)
 {
 	std::uint32_t v = 0;
-	TEST_EQUAL(aux::flz_sw(v), 31);
-	TEST_EQUAL(aux::flz_hw(v), 31);
-	TEST_EQUAL(aux::flz(v), 31);
+	TEST_EQUAL(aux::find_last_zero_sw(v), 31);
+	TEST_EQUAL(aux::find_last_zero_hw(v), 31);
+	TEST_EQUAL(aux::find_last_zero(v), 31);
 
 	v = 0xffffffff;
-	TEST_EQUAL(aux::flz_sw(v), -1);
-	TEST_EQUAL(aux::flz_hw(v), -1);
-	TEST_EQUAL(aux::flz(v), -1);
-
-	v = 0xff00ff00;
-	TEST_EQUAL(aux::flz_sw(v), 23);
-	TEST_EQUAL(aux::flz_hw(v), 23);
-	TEST_EQUAL(aux::flz(v), 23);
-
-	v = 0xff0fff00;
-	TEST_EQUAL(aux::flz_sw(v), 19);
-	TEST_EQUAL(aux::flz_hw(v), 19);
-	TEST_EQUAL(aux::flz(v), 19);
+	TEST_EQUAL(aux::find_last_zero_sw(v), -1);
+	TEST_EQUAL(aux::find_last_zero_hw(v), -1);
+	TEST_EQUAL(aux::find_last_zero(v), -1);
+
+	v = aux::host_to_network(0xff00ff00);
+	TEST_EQUAL(aux::find_last_zero_sw(v), 31);
+	TEST_EQUAL(aux::find_last_zero_hw(v), 31);
+	TEST_EQUAL(aux::find_last_zero(v), 31);
```

Comment:
yes, it's not as simple, actually, `count_leading_zeros` also works with the index from left-to-right, and I think it makes sense, since when you work with `span`, it's not clear the concept of endianness/bit-order to the entire buffer that seems to put the bytes as a string of bits.